pkg/cluster/px/px.go

```diff
@@ -217,12 +229,19 @@ func (ops *pxClusterOps) Upgrade(newSpec *apiv1alpha1.Cluster, opts *UpgradeOpti
 
 	logrus.Infof("Upgrading px cluster to %s. Upgrade opts: %v App drain requirement: %v", newOCIMonVer, opts, isAppDrainNeeded)
 
+	logrus.Info("Attempting to parse kvdb info from Portworx daemonset")
+	_, _, _, platform, configParseErr := ops.parseConfigFromDaemonset()
```

Comment:
nit: I would rather have platform as a variable inside the pxClusterOps struct so that you don't have to pass it around.

Reply:
done